File: generics/generics.cpp
#include <iostream>

using namespace std;

template <class T>
T max_generico(T x, T y) {
	return x > y ? x : y;
}

int main() {

	cout << "Max (5, 6): " << max_generico(5, 6) << endl;
	cout << "Max (6.0, 5.0): " << max_generico(6.6, 5.5) << endl;
	return 0;
}

File: generics/Generics.java
public class Generics {

	public static <T extends Comparable<T>> T max_generico(T x, T y) {
		return x.compareTo(y) > 0 ? x : y;
	}

	public static void main(String args[]) {
		System.out.println("Max (5, 6):" + max_generico(5, 6));
		System.out.println("Max (6.6, 5.5):" + max_generico(6.6, 5.5));
	}
}
